service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/realm/JwtAuthenticatingRealm.java

```diff
@@ -190,6 +190,9 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken authent
     private String extractExternalId(String jwt) {
         final String id;
         try {
+            if (jwtProcessor == null) {
+                init();
```

Comment:
Hi @zanpizmoht , just a question about this init, are you sure we need it? The `onInit()` method of a shiro realm is called while initialising all the realms, so if the JwtAuthenticationRealm is declared in the `shiro.ini` file (as it is in our case), this method should also be called. Thus we should never find the `jwtProcessor` set to `null`.

Reply:
Hey Gianluca, I remember I had problems with this because the jwtProcessor was not initialised. So I discussed this with Alberto and he said I can just add this checker to prevent the jwtProcessor being null. I will double check again if this is needed now (maybe something changed) and will get back to you.

Reply:
I have checked it and those lines are not needed as you said so I removed them now.